src/ecoscope-workflows-core/ecoscope_workflows_core/tasks/io/_set_connection.py

```diff
@@ -1,15 +1,54 @@
-from typing import Annotated
+from dataclasses import dataclass
+from typing import Annotated, Literal
 
 from pydantic import Field
 
 from ecoscope_workflows_core.decorators import task
 
 
+@dataclass(frozen=True)
+class EarthRangerConnection:
+    name: str
+    connection_type: Literal["EarthRanger"] = "EarthRanger"
+
+
+@dataclass(frozen=True)
+class SMARTConnection:
+    name: str
+    connection_type: Literal["SMART"] = "SMART"
+
+
+@dataclass(frozen=True)
+class GoogleEarthEngineConnection:
+    name: str
+    connection_type: Literal["GoogleEarthEngine"] = "GoogleEarthEngine"
+
+
+@task
+def set_er_connection(
+    EcoscopeWorkflowsNamedConnection: Annotated[
+        EarthRangerConnection,
+        Field(description="A named connection.", title="Name"),
+    ],
+) -> EarthRangerConnection:
+    return EcoscopeWorkflowsNamedConnection
+
+
+@task
+def set_smart_connection(
+    EcoscopeWorkflowsNamedConnection: Annotated[
+        SMARTConnection,
+        Field(description="A named connection.", title="Name"),
+    ],
+) -> SMARTConnection:
+    return EcoscopeWorkflowsNamedConnection
+
+
```

Comment:
```suggestion
@task
def set_er_connection(
    connection: Annotated[
        EarthRangerConnection,
        Field(description="A named connection.", title="Name"),
    ],
) -> EarthRangerConnection:
    return EcoscopeWorkflowsNamedConnection


@task
def set_smart_connection(
    connection: Annotated[
        SMARTConnection,
        Field(description="A named connection.", title="Name"),
    ],
) -> SMARTConnection:
    return EcoscopeWorkflowsNamedConnection


```

I think we should consider adapting this pattern to allow for passing this value as a string, rather than an object

https://github.com/wildlife-dynamics/ecoscope-workflows/blob/8f4246440f74adf5da85d1afe44450a28adb9f20/src/ecoscope-workflows-ext-ecoscope/ecoscope_workflows_ext_ecoscope/connections.py#L95-L101